### PR TITLE
[fix]: [PIE-6214]: Corrected Get call for pipeline while getting inputSet for execution, so that it includes branch in find query

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
Issue: Runtime Inputs were not being showed while viewing execution History because get call for pipeline in default branch was failing in get flow for inputsets.
Change: get call for pipeline was failing as it was searching for pipeline in default branch of repo, now get call has been changes to search in the required branch.
Testing: Expected behaviour observed on local for oldGitSync , newGitExperience and Inline cases. Also it my chnages have passes following automation tests. 
Screenshot 2022-11-22 at 2 44 42 PM

## Checklist
- [ ] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)

### CODE REVIEW

**Comment by senjucanon2** [comment]
trigger messagecheck

**Comment by senjucanon2** [comment]
trigger messagecheck

**Comment by senjucanon2** [comment]
trigger ti4

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java` line 418 at r2](https://reviewable.io/reviews/harness/harness-core/40416#-NHOuH8u0RHIBfGHJIB_:-NHOuH8u0RHIBfGHJIBa:buk530j) ([raw file](https://github.com/harness/harness-core/blob/e3ac616cdc43951860e2ef37bbacfd1107bbf8d7/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java#L418)):*
> ```Java
> 
>       EntityGitDetails entityGitDetails =
>           pmsGitSyncHelper.getEntityGitDetailsFromBytes(executionSummaryEntity.getGitSyncBranchContext());
> ```

what if entityGitDetails is null?

**Comment by sahilhindwani-harness** [comment]
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java` line 432 at r2](https://reviewable.io/reviews/harness/harness-core/40416#-NHOuNDP0X9KPpYyUUcr:-NHOuNDP0X9KPpYyUUcs:b-49zwie) ([raw file](https://github.com/harness/harness-core/blob/e3ac616cdc43951860e2ef37bbacfd1107bbf8d7/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java#L432)):*
> ```Java
>       }
>       if (EmptyPredicate.isEmpty(template) && EmptyPredicate.isNotEmpty(yaml)) {
>         template = latestTemplate;
> ```

I think this is wrong, let's just remove this if altogether

**Review by sahilhindwani-harness — CHANGES_REQUESTED** [review]
> Reviewed 1 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
trigger ss

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 3 file(s) to review. 1 change requests. 2. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java` line 418 at r2](https://reviewable.io/reviews/harness/harness-core/40416#-NHOuH8u0RHIBfGHJIB_:-NHPVvHH-FzmWCmC6V9h:b-cbgkn3) ([raw file](https://github.com/harness/harness-core/blob/e3ac616cdc43951860e2ef37bbacfd1107bbf8d7/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java#L418)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

what if entityGitDetails is null?
</blockquote></details>

Have added a null check for it
___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java` line 432 at r2](https://reviewable.io/reviews/harness/harness-core/40416#-NHOuNDP0X9KPpYyUUcr:-NHPVxg9-5sDFc_u0OvR:b-k90mqx) ([raw file](https://github.com/harness/harness-core/blob/e3ac616cdc43951860e2ef37bbacfd1107bbf8d7/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java#L432)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

I think this is wrong, let's just remove this if altogether
</blockquote></details>

I have just removed the redundant part of code, didn't wanted to change the backward compatibility for which this was introduced, may cause regressions

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 3 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 2 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Comment by senjucanon2** [comment]
Closing this as will be removing this get call altogether, after UI had made it changes.

File: pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java [hunk 1]
```diff
@@ -442,6 +442,32 @@ public InputSetYamlWithTemplateDTO getInputSetYamlWithTemplate(String accountId,
...
+    String latestTemplate;
+    if (entityGitDetails != null) {
+      // will come here if the pipeline was remote
+      GitSyncBranchContext gitSyncBranchContext = GitSyncBranchContext.builder()
+                                                      .gitBranchInfo(GitEntityInfo.builder()
+                                                                         .branch(entityGitDetails.getBranch())
+                                                                         .repoName(entityGitDetails.getRepoName())
+                                                                         .build())
+                                                      .build();
+      try (PmsGitSyncBranchContextGuard ignored = new PmsGitSyncBranchContextGuard(gitSyncBranchContext, true)) {
```
Comment (namanvermah) [inline pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java:458]: you dont need guard here, the method you're calling in the guard creates its own guard. Can you check

**Review by namanvermah — APPROVED** [review]
> Reviewed 1 of 1 files at r2, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 1. unresolved discussions.

**Review by namanvermah — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 1. unresolved discussions.

**Review by senjucanon2 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40416)* status: 1. unresolved discussions. 

___
*[`pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java` line 458 at r5](https://reviewable.io/reviews/harness/harness-core/40416#-NHODxuiigD9q0eQjmbq-r5-458:-NHc9WtN2YE684VEgAIH:b-y60xen) ([raw file](https://github.com/harness/harness-core/blob/1e350ea3b46542971b9909f2bfd186a3f932eb69/pipeline-service/service/src/main/java/io/harness/pms/plan/execution/service/PMSExecutionServiceImpl.java#L458)):*
<details><summary><i>Previously, namanvermah (Naman Verma) wrote…</i></summary><blockquote>

you dont need guard here, the method you're calling in the guard creates its own guard. Can you check
</blockquote></details>

I check for some case guard is not set

**Pull request merged**